metrics-collectd/src/test/java/com/codahale/metrics/collectd/Receiver.java

```diff
@@ -0,0 +1,81 @@
+package com.codahale.metrics.collectd;
+
+import org.collectd.api.Notification;
+import org.collectd.api.ValueList;
+import org.collectd.protocol.Dispatcher;
+import org.collectd.protocol.UdpReceiver;
+import org.junit.rules.ExternalResource;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+public final class Receiver extends ExternalResource implements Runnable {
+
+    private int port;
+    private UdpReceiver receiver;
+    private Thread receiverThread;
+    private BlockingQueue<ValueList> queue;
+
+    public Receiver(int port) {
+        this.port = port;
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        queue = new LinkedBlockingQueue<ValueList>();
+        receiver = new UdpReceiver(new QueueDispatcher(queue));
+        receiver.setPort(port);
+        start();
+    }
+
+    private void start() {
+        receiverThread = new Thread(this);
```

Comment:
Instead of implementing `Runnable` by `Receiver`, we could pass the action directly to the `Thread` as a lambda expression.
```java
receiverThread = new Thread(() -> {
            try {
                receiver.listen();
            } catch (Exception e) {
                e.printStackTrace();
            }
        });
```